io-engine/tests/resource_stats.rs

```diff
@@ -0,0 +1,209 @@
+
+    ms_nex.lock().await.stats.reset_io_stats(()).await.unwrap();
+    ms_0.lock().await.stats.reset_io_stats(()).await.unwrap();
+    ms_1.lock().await.stats.reset_io_stats(()).await.unwrap();
+
+    let pool_0_stat = ms_0
+        .lock()
+        .await
+        .stats
+        .get_pool_io_stats(ListStatsOption {
+            name: None,
+        })
+        .await
+        .unwrap();
+
+    let pool_1_stat = ms_1
+        .lock()
+        .await
+        .stats
+        .get_pool_io_stats(ListStatsOption {
+            name: None,
+        })
+        .await
+        .unwrap();
+
+    let nexus_stat = ms_nex
+        .lock()
+        .await
+        .stats
+        .get_nexus_io_stats(ListStatsOption {
+            name: None,
+        })
+        .await
+        .unwrap();
+
+    let nexus_stat: &Vec<IoStats> = &nexus_stat.get_ref().stats;
+    let pool_0_stat: &Vec<IoStats> = &pool_0_stat.get_ref().stats;
+    let pool_1_stat: &Vec<IoStats> = &pool_1_stat.get_ref().stats;
+    let n_stat = nexus_stat.get(0).unwrap();
+    let pool_0_stat = pool_0_stat.get(0).unwrap();
+    let pool_1_stat = pool_1_stat.get(0).unwrap();
+    assert_eq!(n_stat.num_read_ops, 0);
+    assert_eq!(pool_0_stat.num_write_ops, 0);
+    assert_eq!(pool_1_stat.num_write_ops, 0);
+    info!("Stat reset done!");
+    // Run FIO with okay data size.
+    test_fio_to_nexus(
+        &nex_0,
+        FioBuilder::new()
+            .with_job(
+                FioJobBuilder::new()
+                    .with_runtime(10)
+                    .with_bs(4096)
+                    .with_iodepth(8)
+                    .with_size(DataSize::from_mb(REPL_SIZE))
+                    .build(),
+            )
+            .with_verbose_err(true)
+            .build(),
+    )
+    .await
+    .unwrap();
+    info!("Fio completed!");
+    // Checks stats after Fio run.
+    // TODO: This is just an initial validtion to see if stats change after FIO
+    // run. Add more validation later.
+
+    let pool_0_stat = ms_0
+        .lock()
+        .await
+        .stats
+        .get_pool_io_stats(ListStatsOption {
+            name: None,
+        })
+        .await
+        .unwrap();
+
+    let pool_1_stat = ms_1
+        .lock()
+        .await
+        .stats
+        .get_pool_io_stats(ListStatsOption {
+            name: None,
+        })
+        .await
+        .unwrap();
+
+    let nexus_stat = ms_nex
+        .lock()
+        .await
+        .stats
+        .get_nexus_io_stats(ListStatsOption {
+            name: None,
+        })
+        .await
+        .unwrap();
+
+    let nexus_stat: &Vec<IoStats> = &nexus_stat.get_ref().stats;
+    let pool_0_stat: &Vec<IoStats> = &pool_0_stat.get_ref().stats;
+    let pool_1_stat: &Vec<IoStats> = &pool_1_stat.get_ref().stats;
+    let n_stat = nexus_stat.get(0).unwrap();
+    let pool_0_stat = pool_0_stat.get(0).unwrap();
+    let pool_1_stat = pool_1_stat.get(0).unwrap();
+    assert_eq!(n_stat.num_read_ops, 0);
```

Comment:
Why is this 0 after fio?

Reply:
Its outdatated now. I think this assert was after reset.